File: objc/Bridge/Bridge/Rectangle.m
#import "Rectangle.h"

@implementation Rectangle

-(instancetype)initWith:(int)x
                      y:(int)y
                  width:(int)width
                 height:(int)height
            graphicsApi:(id<GraphicsAPI>)api
{
    if(self = [super initWithGraphicsAPI:api]) {
        _x = x;
        _y = y;
        _width = width;
        _height = height;
    }
    return self;
}

-(void)draw {
    [self.graphicsApi drawRectangle:_x y:_y width:_width height:_height];
}

@end

File: swift/Bridge/Bridge/Rectangle.swift
import Foundation

public class Rectangle : Shape {
    var x: Int = 0
    var y: Int = 0
    var width: Int = 0
    var height: Int = 0
    
    convenience init(_ x: Int, _ y: Int, _ width: Int, _ height: Int, _ graphicsApi: GraphicsAPI) {
        self.init(graphicsApi)
        self.x = x
        self.y = y
        self.width = width
        self.height = height
    }
    
    override func draw() -> Void {
        self.graphicsApi.drawRectangle(x, y, width, height)
    }
}
